RespiratoryAssistDevices/R4/files/RespiratoryAssistDevicesPrepopulation-0.1.0.cql

```diff
@@ -145,6 +55,102 @@ define RadDeviceRequested:
   else if  "DeviceRequestHcpcsCoding".code.value = 'E0471' then 'E0471'
   else 'null'
 
+define "RequestEncounter": singleton from (
+  [Encounter] encounter
+    where ('Encounter/' + encounter.id) = device_request.encounter.reference.value)
+define "EncounterDate": "RequestEncounter".period.start.value
+define "EncounterParticipants": "RequestEncounter".participant
+
+// Tester and Lab information
+define "RADTestList": [Observation: "RADLaboratoryTestValueSet"]
+define "RADTestListFromF2FEncounter": 
+  "RADTestList" R
+  where EndsWith(R.encounter.reference.value, "RequestEncounter".id)
+define "TestPerformers":   
+  if exists("RADTestListFromF2FEncounter") then "RADTestListFromF2FEncounter"[0].performer
+  else null
+define "Tester":
+  if exists("TestPerformers") then 
+    "TestPerformers" P
+    where P.type = 'Practitioner'
+    return P.display.value
+  else
+    null  
+define "TestLaboratory":
+  if exists("TestPerformers") then 
+    "TestPerformers" P
+    where P.type = 'Organization'
+    return P.display.value
+  else
+    null  
+
```

Comment:
Lab NPI can be retrieved as Organization.identifier where identifier.system = 'http://hl7.org/fhir/sid/us-npi'

Reply:
DTR can't get the FHIR object unless the reference is related to the patient and is directly referred from the DeviceRequest or ServiceRequest for Practitioner.